livekit/src/room/utils/mod.rs

```diff
@@ -0,0 +1,20 @@
+use std::collections::HashMap;
+
+pub fn calculate_changed_attributes(
+    old_attributes: HashMap<String, String>,
+    new_attributes: HashMap<String, String>,
+) -> HashMap<String, String> {
+    let old_keys: Vec<_> = old_attributes.keys().collect();
+    let new_keys: Vec<_> = new_attributes.keys().collect();
+    let all_keys: Vec<_> = old_keys.into_iter().chain(new_keys.into_iter()).collect();
```

Comment:
i think `keys()` already returns an iterator, you might not need to run `collect().into_iter()` on it